docs/samples/Microsoft.ML.Samples/Dynamic/ImageClassification/ResnetV2101TransferLearningTrainTestSplit.cs

```diff
@@ -166,15 +169,44 @@ public static void Example()
                 (elapsed2Ms / 1000).ToString() + " seconds");
         }
 
+        public static int LoadDataIntoBuffer(string path, ref VBuffer<Byte> imgData)
+        {
+            int count = -1;
+            // bufferSize == 1 used to avoid unnecessary buffer in FileStream
+            using (FileStream fs = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1))
+            {
+                long fileLength = fs.Length;
+                if (fileLength > int.MaxValue)
+                    throw new IOException($"File {path} too big to open.");
+                else if (fileLength == 0)
+                {
+                    byte[] _imageBuffer;
+
+                    // Some file systems (e.g. procfs on Linux) return 0 for length even when there's content.
+                    // Thus we need to assume 0 doesn't mean empty.
+                    _imageBuffer = File.ReadAllBytes(path);
+                    count = _imageBuffer.Length;
+                    Console.WriteLine("File length is zero");
+                }
+
+                count = (int)fileLength;
+                var editor = VBufferEditor.Create(ref imgData, count);
+                fs.Read(editor.Values);                
+                imgData = editor.Commit();
+
+                return count;
+            }
+        }
+
         public static IEnumerable<ImageData> LoadImagesFromDirectory(string folder,
             bool useFolderNameasLabel = true)
         {
             var files = Directory.GetFiles(folder, "*",
                 searchOption: SearchOption.AllDirectories);
-
+            VBuffer<Byte> imgData = new VBuffer<byte>();
             foreach (var file in files)
             {
-                if (Path.GetExtension(file) != ".jpg")
+                if (Path.GetExtension(file) != ".JPEG" &&  Path.GetExtension(file) != ".jpg")
```

Comment:
>if (Path.GetExtension(file) != ".JPEG" &&  Path.GetExtension(file) != ".jpg") [](start = 16, length = 77)

is this really true? The last we read on tensorflow github issues it seemed jpeg_decoder operator could work on all image types, can you please verify this? and if true remove this line.

Reply:
yes jpeg_decoder works on all image types. This was just specific to the file extension in the path string. When I use ImageNet dataset, all the images have ".JPEG" extension. So, I had to check for both kind of extensions.

Reply:
I'm sorry but why do we have this condition? If it works on all image types then shouldn't we be checking for all image types instead of just jpeg?

---
In reply to: [329798266](https://github.com/dotnet/machinelearning/pull/4242#discussion_r329798266) [](ancestors = 329798266)

Reply:
I will change it back.

Reply:
Basically the folders don't just have image files, they also have text files and README which will throw an error. That's why we need this condition.